mypy/subtypes.py

```diff
@@ -1167,6 +1167,14 @@ def restrict_subtype_away(t: Type, s: Type, *, ignore_promotions: bool = False)
 def covers_at_runtime(item: Type, supertype: Type, ignore_promotions: bool) -> bool:
     """Will isinstance(item, supertype) always return True at runtime?"""
     item = get_proper_type(item)
+    supertype = get_proper_type(supertype)
+    if (isinstance(item, (ErasedType, DeletedType))
+            or isinstance(supertype, (ErasedType, DeletedType))):
```

Comment:
Hmm I wonder if `ErasedType` instances can be nested inside unions, for example, and still cause a crash. Maybe `erase_type` should just propagate erased types unmodified? I don't think that `DeletedType` can be nested so it seems safe to handle here, though I'm not sure we need special casing for it. Did you find a use case where we encounter deleted types here, or is it here more as a defensive measure?

Reply:
Hm, good idea. I will try it!